ComplementaryScripts/modelCuration/mapKEGGID.m

```diff
@@ -0,0 +1,275 @@
+
+%only retain mapped metKEGGID if model.metKEGGID is empty
+empties = find(cellfun('isempty',xref_metKEGG(:,1)));
+xref_metKEGG(empties,:) = [];
+empties = find(~cellfun('isempty',xref_metKEGG(:,4)));
+xref_metKEGG(empties,:) = [];
+
+%add metKEGGID into model
+[~,idx] = ismember(xref_metKEGG(:,2),model.mets);
+idx = idx(idx~=0);
+
+for i = 1:length(idx)
+    if ~contains(xref_metKEGG(i,1),'G') %metKEGGID containing G not added, since it is not compatible with current SBML format
+        model.metKEGGID(idx(i)) = xref_metKEGG(i,1);
+    end
+end
+
+%map metMetaNetXID to metMetaCycID via function mapIDsViaMNXref, then map metMetaCycID to metKEGGID
+xref_metMetaCyc = mapIDsViaMNXref('mets',model.metMetaNetXID,'MetaNetX','MetaCyc');
+xref_metMetaCyc(:,2) = model.mets; %match metMetaCycID with model.mets
+xref_metMetaCyc(:,3) = model.metMetaNetXID; %match metMetaCycID with model.metMetaNetXID
+xref_metMetaCyc(:,4) = model.metKEGGID; %match metMetaCycID with model.metKEGGID
+empties = find(cellfun('isempty',xref_metMetaCyc(:,1)));
+xref_metMetaCyc(empties,:) = [];
+
+%load metaCycMets.mat via getMetsFromMetaCyc from RAVEN directory
+MetaCyc_metInfo = getMetsFromMetaCyc('..RAVEN/metacyc/external');
+
+%Input metKEGGID into xref array
+for i = 1:length(xref_metMetaCyc)
+    if isempty(xref_metMetaCyc{i,4})
+        idx = find(ismember(MetaCyc_metInfo.mets,xref_metMetaCyc(i,1)));
+        if idx~=0
+            if ~isempty(MetaCyc_metInfo.keggid{idx})
+                xref_metMetaCyc(i,5) = MetaCyc_metInfo.keggid(idx);
+            end
+        end
+    end
+end
+
+if size(xref_metMetaCyc,2) == 5
+    %only retain mapped metKEGGID if model.metKEGGID is empty
+    empties = find(cellfun('isempty',xref_metMetaCyc(:,5)));
+    xref_metMetaCyc(empties,:) = [];
+    empties = find(~cellfun('isempty',xref_metMetaCyc(:,4)));
+    xref_metMetaCyc(empties,:) = [];
+    
+    %add metKEGGID into model
+    [~,idx] = ismember(xref_metMetaCyc(:,2),model.mets);
+    idx = idx(idx~=0);
+    
+    for i = 1:length(idx)
+        if ~contains(xref_metMetaCyc(i,5),'G') %metKEGGID containing G not added, since it is not compatible with current SBML format
+            model.metKEGGID(idx(i)) = xref_metMetaCyc(i,5);
+        end
+    end
+else
+    xref_metMetaCyc(:) = []; %erase all cell array, since no KEGGID matched and added
+end
+
+%map metMetaNetXID to metSEEDID via function mapIDsViaMNXref, then map metSEEDID to metKEGGID
+xref_metSEED = mapIDsViaMNXref('mets',model.metMetaNetXID,'MetaNetX','SEED');
+xref_metSEED(:,2) = model.mets; %match metSEEDID with model.mets
+xref_metSEED(:,3) = model.metMetaNetXID; %match metSEEDID with model.metMetaNetXID
+xref_metSEED(:,4) = model.metKEGGID; %match metSEEDID with model.metKEGGID
+empties = find(cellfun('isempty',xref_metSEED(:,1)));
+xref_metSEED(empties,:) = [];
+
+temp = what('ModelSEEDDatabase\Biochemistry'); %require cloned/downloaded GitHub repo ModelSEEDDatabase
```

Comment:
the same comment as before in line 97

Reply:
Resolved in b8a5d3283c45e15a654eb5e89e91570a7406ff61.